app/jobs/iiif_print/jobs/create_relationships_job.rb

```diff
@@ -103,25 +113,50 @@ def create_relationships(parent:, ordered_children:)
               @errors << e
             end
           end
-          parent.save! if @parent_record_members_added && @number_of_failures.zero?
+
+          if @parent_record_members_added && @number_of_failures.zero?
+            if parent.respond_to?(:save!)
```

Comment:
This switching logic could be extracted to the adapter logic:  

- lib/iiif_print/persistence_layer/active_fedora_adapter.rb
- lib/iiif_print/persistence_layer/valkyrie_adapter.rb

Create a class method `save!(resource:)` and have the active fedora adapter implement the logic in the `if` portion; and the valkyrie_adapter implement the logic in the `else` portion.